modules/scalacheck/shared/src/test/scala/eu/timepit/refined/scalacheck/StringArbitrarySpec.scala

```diff
@@ -26,4 +31,6 @@ class StringArbitrarySpec extends Properties("StringArbitrary") {
   property("FiniteString[10]") = checkArbitraryRefinedType[FiniteString[W.`10`.T]]
 
   property("Size[Equal[8]]") = checkArbitraryRefinedType[String Refined Size[Equal[W.`8`.T]]]
+
+  property("NonEmptyFiniteString[10]") = checkArbitraryRefinedType[NonEmptyFiniteString[W.`10`.T]]
```

Comment:
This fails to compile due to diverging implicit resolution:

```scala
/Users/hjdskes/github/refined/modules/scalacheck/shared/src/test/scala/eu/timepit/refined/scalacheck/StringArbitrarySpec.scala:35:67: diverging implicit expansion for type org.scalacheck.Arbitrary[(eu.timepit.refined.collection.NonEmpty, eu.timepit.refined.collection.MaxSize[Int(10)])]
[error] starting with method arbTuple2 in trait ArbitraryArities
[error]   property("NonEmptyFiniteString[10]") = checkArbitraryRefinedType[NonEmptyFiniteString[W.`10`.T]]
```

Do I need to define a new method in `scalacheck/string.scala` to get this to work?

Reply:
Maybe it compiles if `NonEmptyFiniteString` is defined as `String Refined Size[Interval.Closed[_1, N]]`.

Reply:
Yep, it does!